### DISCUSSION TITLE
Refrigerant Impact Tool

### DISCUSSION THREAD

**Discussion by kayleighhoude** [BHoM/BHoM#1405] [Q&A]
This will serve as a discussion thread for comments, questions and suggestions related to the Refrigerant Impact Tool

**Comment by jonyoung1216** [comment]
Hello, I'm having trouble accessing the Tool through the website (https://www.refrigerantimpact.org/). Regardless of which browser I use, I get the following error  messages:
"The app is loading... If it is your first time opening this app on your machine, this might take a bit longer." 
Then a flag at the bottom pops up with: "An unhandled error has occured. Reload."

I've tried waiting 30min+ for the page to load, as well as reloading the page, but I keep running into the same issue.

Is anyone else experiencing this? Is there something else I should try to access the tool?

Thanks!

**Reply by adecler** [reply]
Hi @jonyoung1216 , 

What browser are you using ? If you are using Internet Explorer, I would recommend switching to something else as it is not supporting any modern browser feature. I have tested the website with Chrome, Edge and Firefox so one of those three would be a good alternative.

I have raised an issue to add a message to Internet Explorer users.

**Reply by jonyoung1216** [reply]
Hi @adecler, thanks, I appreciate the response.
I'm actually getting this error on both Chrome and Edge (the only 2 browsers I currently have). That's interesting to know that it is working for you on these browsers. Maybe there are some browser settings that are causing the error? I will try some things and let you know if anything resolves it. Let me know if you have any thoughts or ideas to try.

Thanks again

**Reply by adecler** [reply]
Hi @jonyoung1216 ,

Could you press `F12` once you get the error popup and share the content of the console? 

For reference, mine looks like this on Chrome: 
![image](https://user-images.githubusercontent.com/16853390/179015417-b1736bd1-9fe9-4d43-894e-05fd9c9885b7.png)

**Comment by kelsworth** [comment]
A few members of the KieranTimberlake Research Group reviewed the tool and our comments are tabulated below. First, we appreciate Buro Happold’s effort and contribution in accounting for the refrigerant’s carbon impact! Thank you! Second, we are examining this tool through the perspective of architects, so we do not have deep knowledge on the datasets, but we are also reviewing this as Tally developers. Hopefully these are helpful comments to improve legibility and broaden the audience for this tool!

- Overall, we would love to see this tool used for design exploration and optioning. Can a future version include an iteration feature so we can study side-by-side options?

- Would it be possible to provide the quantitative assumptions for life expectancy and leakage? For instance, how many years are assumed for the equipment life expectancy. Could this be displayed to the side of the input text?

- Would alternative units for refrigerant charge be useful? Can this list include other units commonly used by engineers, globally?

- Under the LEED Life Expectancy dataset, specific equipment are listed. Is there a way to add more to this list, such as WSHP or VRF?

- We think it would be helpful to report total GWP over the building life so it can be easily used alongside LCA reports. Would you be able to add an input for building life (typically 60 years) that would expand the refrigerants impact beyond it's independent system life? 

- On the outputs frame, “Annual (X years)” caused some confusion in terms of interpretation. It wasn’t until we iterated through different HVAC systems we noticed that this the total leakage through the system life and not the annual rate of leakage. Can you please provide a header for this column of data that defines these parameters?

- We also noticed that different datasets have different GWPs. Using R-410a as an example, we found different total GWP using ASHRAE and GHG. This is likely due to the emissions rate being 725 for R-410 under the GHG Protocol dataset. This may need to be corrected but overall, some guidance on which dataset to use, if the results vary, would be helpful.

- Minor bugs: The link to the CISBE document is broken and the ASHRAE report is behind a paywall.

**Reply by kayleighhoude** [reply]
Hi Kit! Thanks for the thorough review, much appreciated!

1) The idea of adding an optioneering feature sounds great, do you think this would ideally take the form of being able to save out results or to be able to run multiple selections in parallel?
2) Equipment life expectancy and leakage are all rendered in the results (the GHG/LEED datasets lead to the life expectancy in years on the annual calculation and the % loss is rendered on the right side, depending on which dataset you choose some include installation and EOL, others do not)
![image](https://user-images.githubusercontent.com/34316256/184008164-44c14c6a-8fdc-495b-b493-cbe0263febe7.png)

3) We stuck with kg for the charge value as most LCA studies are completed in metric - that 500kgCO2/m2 figure is one that comes to mind, not to mention that BHoM is all in metric. We did make an exception for non-metric units in the equivalencies portion (litres of petrol is not the same as gallons of gasoline!)
4) That is correct in that annual is the annual sum rather than the individual annual amount, see point 2 also (maybe some more clarity around this in the about section would help)
5) We provided as many datasets as possible (those provided by MEP2040 especially) the values do vary, hence the provision of the links to the original sources - we figured people would select datasets they trusted by name (CIBSE for UK folks, ASHRAE for US) regardless of their variation in values (which is unfortunate! especially for the drastic annual loss variation across datasets)
6) ASHRAE 228P is behind a paywall unfortunately as you have pointed out, the CIBSE link appears to be broken on their end as well (see here: https://www.cibse.org/policy-insight/news/new-embodied-carbon-calculator-for-building-services)

**Comment by jneb802** [comment]
Hello, I'm building a refrigeration monitoring tool and am interested in using this to calculate the climate impact my users are having based on their refrigerants. The [source code link](https://github.com/BuroHappoldEngineering/RefrigerantImpact_Tool) isn't working anymore, is there a new link you could provide?

**Reply by kayleighhoude** [reply]
Hello- apologies but the source code resides in a closed source repository
